pkg/database/token.go

```diff
@@ -250,15 +261,43 @@ func (db *Database) VerifyCodeAndIssueToken(t time.Time, authApp *AuthorizedApp,
 			return ErrVerificationCodeUsed
 		}
 
-		if _, ok := acceptTypes[vc.TestType]; !ok {
+		if _, ok := request.AcceptTypes[vc.TestType]; !ok {
 			db.logger.Debugw("checked not of accepted testType", "ID", vc.ID)
 			return ErrUnsupportedTestType
 		}
 
+		// Check associated UserReport record if necessary.
+		if vc.UserReportID != nil {
```

Comment:
This code is definitely clear, but wouldn't it be more performant to run:

```
UPDATE user_repots SET code_claimed = TRUE WHERE code_claimed = FALSE AND nonce = ?
```

And then return NotFound if the number of rows modified == 0?

Reply:
That also has the advantage of being totally atomic without locking.

Reply:
done